benches/criterion_stark.rs

```diff
@@ -0,0 +1,55 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+
+mod functions;
+mod util;
+
+pub fn proof_benchmarks(c: &mut Criterion) {
+    let mut group = c.benchmark_group("STARK");
+    group.sample_size(10);
+
+    for trace_length in [8, 128, 256, 512, 1024].iter() {
+        group.throughput(Throughput::Bytes(*trace_length as u64));
+        group.bench_with_input(
+            BenchmarkId::new("Simple Fibonacci", trace_length),
+            trace_length,
+            |b, &t| {
+                b.iter(|| functions::stark::prove_fib(t));
```

Comment:
We should use `criterion::black_box` on these benchmarks. The compiler sometimes over-optimizes these kinds of cases, and so these benches may report a lot less time than they should.

```suggestion
                b.iter(|| functions::stark::prove_fib(black_box(t)));
```